wazo_test_helpers/asset_launching_test_case.py

```diff
@@ -346,25 +381,79 @@ def _maybe_dump_docker_logs(cls):
             logger.debug('Container logs dumped to %s', logfile.name)
 
     @staticmethod
-    def get_log_directory():
+    def get_log_directory() -> str:
         if not AssetLaunchingTestCase.log_dir:
-            char_set = string.ascii_lowercase
-            default_logging_dir = '/tmp/wazo-integration-{}'.format(
-                ''.join(random.choice(char_set) for _ in range(8))
+            random_strings = "".join(
+                random.choice(string.ascii_lowercase) for _ in range(8)
             )
             AssetLaunchingTestCase.log_dir = os.getenv(
-                'WAZO_TEST_DOCKER_LOGS_DIR', default_logging_dir
+                'WAZO_TEST_DOCKER_LOGS_DIR', f'/tmp/wazo-integration-{random_strings}'
             )
             if not os.path.exists(AssetLaunchingTestCase.log_dir):
                 os.makedirs(AssetLaunchingTestCase.log_dir, mode=0o755)
-        return AssetLaunchingTestCase.log_dir
+        return str(AssetLaunchingTestCase.log_dir)
 
 
-def _run_cmd(cmd, stderr=True):
-    logger.debug('%s', cmd)
-    stderr = subprocess.STDOUT if stderr else None
+class AssetLaunchingTestCase(AbstractAssetLaunchingHelper, unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.launch_service_with_asset()
 
-    completed_process = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=stderr)
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.stop_service_with_asset()
+
+
+def make_asset_fixture(
+    asset_class: type[AbstractAssetLaunchingHelper],
+) -> Callable[[], Generator[type[AbstractAssetLaunchingHelper], None, None]]:
+    """
+    Helper to create asset loading fixtures for Pytest in conftest.py or test files.
+
+    Usage:
+    ```python
+    class MyAssetClass(AbstractAssetLaunchingHelper):
+        assets_root = Path(__file__).parent / '..' / '..' / 'assets'
+        asset = 'asset-name'
+        service = 'my-service'
+
+    # Note: assignment is important to be detected by pytest.
```

Comment:
I suppose, I don't know why they do it that way. Although, since decorators can only be used via `@` on a function definition, you would have to pass it manually to the decorator or redefine the entire function.

Either:
```
pytest.fixture(name='my_asset_asset')(make_fixture(BaseAssetHelper))
```
which is clearer, but pretty ugly or essentially copy and pasting the same method in every class and just changing the name and class.

Reply:
Would something like this work for you? I'm fine with having to repeat that once for each asset class of one repo.

```python
@pytest.fixture
def base_asset():
    yield from make_fixture(BaseAssetHelper)
```